File: Marvellous_LB/2-Problems on digit/program9.c
#include<stdio.h>

int Maximum(int iVal)
{
    int iDigit=0;
    int iStored=0;

    if(iVal<0)
    {
        iVal=-iVal;
    }

    while(iVal!=0)
    {
        iDigit=iVal%10;

// Time complexity
        if(iDigit==9)
        {
            iStored=iDigit;
        }
////????????????????????????????   
      else if(iStored<iDigit)
      {
        iStored=iDigit;
      }
         iVal=iVal/10;
    }

    return iStored;
}

int main()
{
    int iNo=0;
    printf("Enter a number to check its largest:\n");
    scanf("%d",&iNo);

    int iRet=Maximum(iNo);

    printf("The largest in %d is:\n%d",iNo,iRet);

    return 0;
}

File: Marvellous_LB/8-JAVA/program9.java
import java.util.*;
class Numbers
{
    public int FactorsMult(int iVal)
    {
        int iMult=1;
        int iCnt=0;
        for(iCnt=1;iCnt<=(iVal/2);iCnt++)
        {
            if(iVal%iCnt==0)
            {
                iMult*=iCnt;
            }
        }
        return iMult;
    }
}
public class program9
{
    public static void main (String arg[])
    {
        int iNo=0,iRet=0;
        Scanner sobj=new Scanner(System.in);

        System.out.println("Enter number to check even Factors : ");
        iNo=sobj.nextInt();

        Numbers nobj=new Numbers();
        iRet=nobj.FactorsMult(iNo);

        System.out.println("Multiplication of factors is : "+iRet);
    }
}
